flake.nix

```diff
@@ -17,32 +26,160 @@
+              supportedFeatures = [ "benchmark" "big-parallel" ];
+            }];
+            extraOptions = ''
+              experimental-features = nix-command flakes
+            '';
           };
         };
+        global = { pkgs, ... }: {
+          documentation.man.enable = false;
+          nixpkgs.overlays = [ overlay ];
+          system.stateVersion = "22.05";
+          networking.firewall.allowedTCPPorts = [ port ];
+          systemd.services.binplz-server = {
+            enable = true;
+            description = "Binplz binary server";
+            wantedBy = [ "multi-user.target" ];
+            script = ''
+              ${pkgs.binplz-server}/bin/binplz-server \
+                --port ${builtins.toString port} \
+                --program-db ${inputs.nixos-channel}/programs.sqlite
+            '';
+            serviceConfig = {
+              Restart = "always";
+              Type = "simple";
+              Environment = "PATH=\"/run/current-system/sw/bin:$PATH\"";
+            };
+          };
+          users = {
+            mutableUsers = false;
+            users.root = {
+              password = "nixos";
+              openssh.authorizedKeys.keys =
+                builtins.concatMap
+                  (x: if builtins.isList x then x else [ ])
+                  (builtins.split
+                    "(ssh-[^\n]*)\n"
+                    (builtins.readFile ./secrets/trustees));
+            };
+          };
+          services = {
+            openssh.enable = true;
+            openssh.permitRootLogin = "prohibit-password";
+          };
+        };
+      };
+
+      server-shell = let hspkgs = pkgs.haskellPackages; in
+        hspkgs.shellFor {
+          withHoogle = true;
+          packages = p: [ p.binplz-server ];
+          buildInputs = [
+            hspkgs.cabal-install
+            hspkgs.haskell-language-server
+            hspkgs.hlint
+            hspkgs.ormolu
+            pkgs.bashInteractive
+            pkgs.rlwrap
+            pkgs.sqlite
+            pkgs.age
+          ];
+          BINPLZ_NIX_PROGRAM_DB = "${inputs.nixos-channel}/programs.sqlite";
+        };
+
+      deploy-shell = pkgs.mkShell rec {
+        packages = [
+          pkgs.terraform
+          pkgs.awscli2
+          pkgs.rlwrap
+          pkgs.sqlite
+          pkgs.age
+        ];
+        TF_VAR_binplz_ami_path = "${inputs.self.packages.${system}.server-ami}/${image-name}.vhd";
+        # Image name on AWS. The purpose is to invalidate when the image contents change.
+        # TODO cleaner way of getting the store path than `rec`
+        TF_VAR_binplz_image_name_hash =
+          let hash = builtins.substring 11 8 TF_VAR_binplz_ami_path;
+          in "${image-name}-${hash}";
+      };
+
+      encrypt_secrets = pkgs.writeShellApplication {
+        name = "encrypt_secrets";
+        runtimeInputs = [ pkgs.age ];
+        text = ''
+          echo ${builtins.toString ./.}
+        '';
+      };
```

Comment:
What's this for?  I couldn't find anywhere that you're using it.

Reply:
Oh, good catch, this should have been removed.